### ISSUE TITLE
Sprint - Oct 28 to Nov 8

### ISSUE THREAD

**Issue by Phanatic** [PostHog/posthog#25746]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by Phanatic** [comment]
# Team Surveys

## Retro

✅  Beta testing no-code experiments with customers
✅  Global CSS for Surveys
✅ Various small UI fixes for Surveys

### Side Quests 


---

## OKRs

1. Implement Enterprise-ready features for Surveys
2. Increase adoption of Surveys.

### High priority
* Continue beta testing no-code experiments with customers.
*  Custom Survey templates at the team level
* Investigate support for mobile surveys in `posthog-react-native` library

**Comment by pauldambra** [comment]
# Team Everyone comes to Spain

Another 1 week shipping, 1 week travelling sprint 

**Support hero:** @richard-better with @pauldambra supporting     

## Retro

* lots of diversions this sprint, ES6 bundling in posthog-js, rrweb upgrades, support driving a lot of small changes (24 PRs in 2 weeks 🤯), lots of small pieces is pretty tiring @pauldambra 

---

## Mobile

* 🔁 get mobile recordings into billing_usagereport @pauldambra 
    * need a bit more clarity here first
* ✅ track size of mobile recordings so we understand better how expensive mobile ingestion / storage is @pauldambra 
* get first alpha of flutter replay out @marandaneto
* implement the swift ui masking from last sprint's experiment @ioannisj 
* auto capture for element interactions in iOS https://github.com/PostHog/posthog-ios/issues/179 @ioannisj 

## Replay

* ✅ roll out the URL trigger 
* 🔁 and adding a URL denylist @richard-better 

### Low priority / side quests

* 🔁 trigger to start recordings on event @richard-better 
* 🔁 lazy loading replay in the SDK @pauldambra 

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

🟡📱Goal 1: People think of PostHog as a mobile solution
🟡 📹 Goal 2: Replay playback 3000
    - delayed-ish till Nov and Cory time in Valencia
🟡 🎤 Goal3: Ship the things users keep asking for
 
### High priority

## Mobile

* mobile replay billing clarity @pauldambra 

## Replay

* replay URL denylist @richard-better 
* a brief diversion before querying network data - wrap websockets as well as fetch @pauldambra 

### Low priority / side quests

* trigger to start recordings on event @richard-better 
* lazy loading replay in the SDK @pauldambra

**Comment by mariusandra** [comment]
# Team Customers Drive Profits

- Support: Marcus
- Marius off first week
- Ben is back from Tueday

## Retro

- CDP pricing @annaszell @mariusandra 
  - We decided to keep the current pricing structure, and reduce costs slightly. Details still in progress.
- Figure out what to build with messaging (and build v0.0.1-alpha) @mariusandra 
  - Long chat in an RFC. Started actually building. Alpha ready by the end of the sprint.
- Error tracking & support. Cross team support, not a lot of feature work (on CDP). @oliverb123 
  - 
- Google & Meta ads destinations, help with messaging UI @MarconLP 
  - 

## OKRs
* Widespread usage of the new CDP 
* Messaging V1 
* Ingestion is rock solid

## Sprint focus

### High priority

- Get messaging v0 ready for dogfooding. @mariusandra

**Comment by daibhin** [comment]
# Team ~GPS~ Error Tracking

Time off:
@daibhin EU on-call Week 1
@daibhin @oliverb123 Valencia offsite in Week 2
@neilkakkar working on Experiments / off second week

## Retro

🟡 Storing resolved source maps @oliverb123 
- Made good progress but CDP support was busier than expected
- Source map fetching & Postgres modelling in place
- At risk of not finishing reading / writing to S3

🟡 Overrides table for issue merging @daibhin
- Deciding on approach took longer than expected
- Open PRs to create the necessary tables
- Hoping to write new JOIN queries by the end of the week

🟡 _Advanced_ fingerprinting in Rust @neilkakkar 
- First version ported over from plugin server
- Need to add logic to decide on
- Fingerprinting logic (e.g. exclude unresolvable stack frames if trace includes resolvable frames)

⚪️ Add language tag to frames on the JS & Python client @daibhin
✅ Choose persistent key / value store for stack frame resolution (Postgres or Dynamo or something else) @oliverb123 
✅ Rename stack trace properties on JS SDK (beware: Sentry integration relies on some of this) @neilkakkar 

## OKR

🟢 Human readable stack traces for Javascript
🟢 Ensure scalability of query time issue grouping
⚪️ Alerting via CDP
🟢 Launch

### High priority

Stack trace demangling enabled for PostHog team
- Writing / Reading source maps via S3 (Rust) @oliverb123
- In-app source map uploading @daibhin 
  - Button on unresolved frames to add a source map
  - Writing source maps via S3 (Django)
  - Hook up invalidation of frames in Postgres (cascade delete)

Fingerprint to issue lookup flow
- Postgres client / queries @oliverb123
- Wiring up fingerprinting & issue lookup to demangling flow @daibhin

### Low priority / side quests

- Frontend query caching to improve page load speed @daibhin

**Comment by zlwaterfield** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

- @raquelmsmith 
	- [ ] Fix forecasting bug that says peoples' forecast is much higher than what they see on billing page
	- [ ] Fix bugs with product analytics onboarding flow
	- [x] Personless default change
	- [x] Make sure metrics move didn't break wnything
	- [x] Re-up activation metrics for product analytics - try using `insight saved` instead of `insight created`
	- [x] Use new product intent table in activation metrics
- @patricio-posthog 
	- [x] Continue working with the sync invoices changes to use webhooks instead of a daily sync using Stripe API
	- [x] Check billing limit bug raised by jdoodle (support ticket)
	- [ ] May investigate slow load times in billing page and try to optimize them (confirm first if it only affects ElevenLabs)
- @zlwaterfield 
	- RBAC
		- [ ]  Finish v1 of the RFC for billing
		- [ ]  Start working on it - decide if we should continue on Ben's PR or start from scratch and do it piece by piece - get at least 1 PR merged for it
	- [x] Finish up the Technical Exercise RFC - resolve comments and get merged, share out
	- [ ] Get the CC entry PR shipped
	- [x] Adding user id to billing token for events
	- [x] Adding Hogfetti component
	- Growth review (if time permits)
		- [x] We had a number of “accidental” signups that got refunded in sept, which might show up as churn
		- [x] Teams plan survey after they subscribe
		- [ ] Use new ICP data to compare with CC activations - are high ICP people still subscribing at the same rate?
		- [x] Add is subscribed bool and mrr to org group
</details>  

## Q4 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Get to a 4/10 in Enterprise platform support (Zach)
	- 🟡 Release Enterprise addon and iterate with sales to find a pricing model that makes sense
	- 🟡 Build RBAC
2. 🟡 Make revenue & forecasting more reliable (Patricio)
	- 🟡 Rejig how we do sync_invoices so we don’t see big drops - respond to webhooks instead of large syncs
	- ⚪ Evaluate temporal vs celery for usage reports and billing tasks so we can adhere to rate limits better
	- ⚪ Side hustle, if time permits: Rejig how anonymous / identified events are billed, as the current implementation is confusing.
3. 🟡 Activation: reverse trials, analytics onboarding flow, understand CC activation (Raquel)
	- ⚪ Dig into the data, watch replays to figure out why CC activation rate has been falling
	- ⚪ Add reverse trial to onboarding
	- 🟡 Add reverse trial to billing page for teams

## This sprint

Time off:  Raquel (1/2 day)

- @zlwaterfield 
	 - [ ] Support week 1
	 - RBAC
	 	- [ ] Finish RFC for billing and get feedback on it
	 	- [ ] Start working on it - get at least 1 PR merged for it
	- [ ] Get the in-app credit card entry experiment shipped 
	- [ ] Make a few improvements to the technical exercise (based on using it for 2 weeks)
	- [ ] 	Project limit support requests
		- https://posthog.slack.com/archives/C0428JHABK2/p1729272091461489 
		- https://posthoghelp.zendesk.com/agent/tickets/19005 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/22146) 
- @raquelmsmith 
  - [ ] Slowly roll out personless default change, deal with any issues
  - [ ] Fix bugs with product analytics templates onboarding flow
  - [x] Ship session replay "templates" screen
  - [ ] Write data warehouse activation query
  - [ ] Think about how to redo onboarding so people can pick multiple products at once
  - [ ] Respond to customer about projects-on-invites
- @patricio-posthog 
	- [ ] Billing limit: Get this bug fixed once and for all so we can move on
	- [ ] Forecasting bug: Fix forecasting and if it's a good idea, reimplement it with a new version 
	- [ ] Improve current `billing` API endpoints (optimize or spit current ones into separate endpoints)
	- [ ] Check if the webhooks for invoices are working fine (check events at Posthog) / turn off the daily sync to test that.

**Comment by fuziontech** [comment]
# Team 🧊 Data
![image](https://github.com/user-attachments/assets/9b9485f0-2537-481a-9633-1ad17851b64b)

## OKR Q2 2024

### Objective

**Goals for Q4:**

- Improve elasticity and flexibility of our data store by putting all our data in Iceberg
    - [Work with Altinity](https://docs.google.com/document/d/12icx71-_kV0aICP3igtOc0di_yrey_W82HzbadtNPBY/edit?usp=sharing) effectively to ship read path for Iceberg on ClickHouse - Brett Hoerner
    - Setup infrastructure to ship all of our data to Iceberg on S3 - James Greenhill
    - Shipping Query logs to Iceberg - James Greenhill
- Continue improving CH operational expertise
    - Upgrade to a later version of Clickhouse - James Greenhill
    - Capacity planning - James Greenhill
    - Automation - Daniel Escribano
    - Put some of the basic mitigation operations in runbooks - Daniel Escribano
- Schema management
    - Tool for schema migration (coordinator schemas) - Daniel Escribano
    - Tool for long running mutations - Daniel Escribano
- Continued investment in performance
    - Tooling for other teams to understand which queries are slow and why - Ted Kaemming
    - Investigate variability of queries - Ted Kaemming
    - Per-team limits on queries/query complexity (needs product work) - Ted Kaemming

## Offsite week 2 Nov 4->8 in Miami!

## Board
https://github.com/orgs/PostHog/projects/85/views/2

## Retro

### 🔥 Outages 🔥 
- [#inc-2024-10-23-clickhouse-eu-lag](https://posthog.slack.com/archives/C07T7L6FZK6)
Only one incident in nearly 20 days!

### Last Sprint's Goals
- Reformat 🧊 berge schema to accommodate Batch Export work @bretthoerner @Daesgar ✅ 
- Prod writing events to Iceberg to support ^ @Daesgar ✅ 
- Have a plan for Materialized Columns on 🧊  @fuziontech @tkaemming ⚠️ 
- Schedule compaction job in DW/Temporal somewhere @bretthoerner & DW ✅ 
- Setup TTL for events for 48 months to gp3 disk @fuziontech ⚠️ 
- Complete upgrade to CH 24.8 @fuziontech ⚠️ 
- First PRs with Altinity @bretthoerner (most likely) ✅ 
- Automated TF for CH @Daesgar  ✅ 

## This Sprint's Goals 

### TL;DR

- Continue 24.8 upgrade and test fix @fuziontech 
- Work with DW/BE for pulling fresh data from Iceberg table for Batch Exports @bretthoerner 
- Ongoing Delete work @fuziontech 
- Integrate compaction into DW DAG @bretthoerner 
- Nullable performance improvements @tkaemming 
- Query log to Iceberg @fuziontech 
- Setup TTL for events for 48 months to gp3 disk @fuziontech ⚠️ 
- EU to single AZ in eu-central-1c @Daesgar 

#### Side Quest

**Comment by jurajmajerik** [comment]
## Retro
🟢 Feature Success offsite
	- worked on Experiment UI improvements during hackathon
	- Phani walked us through No-code experiments, still some todos before full handover

🟡 Continued on HogQL for Experiments (https://github.com/PostHog/posthog/issues/24914) - about 60-ish % done - query endpoints are now callable

## Sprint goals
- Support hero for the Feature Success queue
- Continue HogQL rewrite
	- plug statistical calculations
	- migrate UI behind a feature flag
- Review the methodology with a statistician - starting this Friday!

## Secondary
- Investigate default exposure for trends showing `1`

**Comment by Twixes** [comment]
# Team AI Product Manager

## Retro

- ✅ ~~Router between AI agent's skills (trends, funnels, SQL, etc.)~~ @skoob13 
- Support for insight types @skoob13 @Twixes Deprioritized in favor of quality
- Basic SQL support @Twixes Deprioritized in favor of quality
- ✅ ~~Answer feedback button~~ @Twixes 
- ✅ ~~Extra based on feedback: Query definition preview~~ @Twixes 

- ~~Environments out as a feature preview~~ @Twixes Rolling over due to support

## Objective

LLM-based product manager as a service

### High priority

- Funnels support @skoob13 
- Tune agent/prompt for more relevant results (primary: excluding most $-prefixed events/properties) @skoob13 @Twixes
- Stream feedback on the agent's current task in conversation @Twixes
- PostHog dashboard for AI product manager use @Twixes
- OpenAI data processing opt-in (to be determined with @fraserhopper) @Twixes 

### Low priority / side quests

- Evaluation framework for understanding if quality of results is increasing @skoob13
- Environments out as a feature preview @Twixes

**Comment by aspicer** [comment]
# Team Product Analytics

**Support hero:** @aspicer and @anirudhpillai 

## Retro

Anirudh

*   [✅](https://emojipedia.org/check-mark-button) Create snoozing mechanic for firing alerts
*   [✅](https://emojipedia.org/check-mark-button) Launch relative alerts
*   [🟡](https://emojipedia.org/large-yellow-circle) Roll out to more users (talk to comms about finding beta test customers)

Sandy

*   [✅](https://emojipedia.org/check-mark-button) UDF rollout for steps + moving funnel trends to UDF
*   [🟢](https://emojipedia.org/large-green-circle) Get a subset of cohorts calculating on HogQL

## Next Sprint

Sandy

*   Support (7 days)
*   Continue porting cohorts 

Thomas (Returns on Nov 4th!)

*   Resume filter to query work?

Anirudh (OOO through Nov 6th)

*   Support (3 days)

**Comment by robbie-c** [comment]
# Team web analytics

**Support hero:** @robbie-c 

## Retro

Made some progress on cookieless tracking, but was out sick for part of it

#### Tasks
Cookieless prototype
- 🟢 front end
- 🔴 back end

## OKR

* Continue to improve performance
* Build a small amount of more advanced marketing features, to measure interest
* Cookieless tracking
* Improve the NextJS experience
* More page-level insights
* Improve the documentation, and link to it more from the app

### High priority
- Finish cookieless prototype
- Onboard Mike / have a successful offsite

**Comment by dmarticus** [comment]
# Team Flags

**Support hero:** mostly @jurajmajerik (old feature success team is sharing the same support queue until our newly split teams scale up)

## Retro

- Didn't get much work done on OKR 1, mostly spent time on OKR 2 (maintaining old decide endpoint)
- spent most of the sprint fixing issues, doing calls with customers, and then fixing more issues.  
- Finally paid down the outstanding support volume and have fixed some of the bugs that users were complaining about, so this sprint I'm hoping to double down on OKR 1 work

### Things shipped
 
- ✅ Fixed a nasty bug with GeoIP not returning city names in `/decide`: https://github.com/PostHog/posthog/pull/25747
- Fixed a fewer less-scary but still embarrassing UI bugs in the flags UI (see the last few pulls here: https://github.com/pulls?q=is:pr+author:dmarticus+archived:false+is:closed)

---

## OKRs

1. Ship new feature flags in production and accomplish the 10x scale + 2x speed boost performance targets (ideally decreases infra spend by N% as well, will try to figure out a more specific target going forward)
2. Maintain current /decide service and make optimizations.

## Sprint

### High priority

- Finish static and dynamic cohort matching for new feature flag service
    - support full subset of cohort match operations (`not_in`, etc) at the service/DB level
- Prepare for Haven's onboarding in Seattle

### Low Priority/Stretch Goal

- Port over existing feature flag integration tests from the django to rust repo (lots of valuable alpha in our existing `/decide` tests, but this doesn't need to happen this sprint)
